sample/android/src/main/java/com/freeletics/flowredux/compose/PopularRepositoriesUi.kt

```diff
@@ -0,0 +1,62 @@
+package com.freeletics.flowredux.compose
+
+import androidx.compose.material.Scaffold
+import androidx.compose.material.SnackbarDuration
+import androidx.compose.material.rememberScaffoldState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.ui.res.stringResource
+import com.freeletics.flowredux.AndroidFlowReduxLogger
+import com.freeletics.flowredux.R
+import com.freeletics.flowredux.sample.shared.*
+import kotlinx.coroutines.launch
+
+@Composable
+fun PopularRepositoriesUi() {
+    val stateMachine = InternalPaginationStateMachine(
+        githubApi = GithubApi(),
+        logger = AndroidFlowReduxLogger
+    )
+
+    val (state, dispatch) = stateMachine.stateAndDispatch()
+
+    // Timber.d("New State ${state.value}")
+    val scaffoldState = rememberScaffoldState()
+
+    SampleTheme {
+        Scaffold(scaffoldState = scaffoldState) {
+            when (val s = state.value) {
+                is LoadFirstPagePaginationState -> LoadingUi()
+                is LoadingFirstPageError -> ErrorUi(dispatch)
+                is ContainsContentPaginationState -> {
+                    val showLoadNextPageUi = s.shouldShowLoadMoreIndicator()
+                    val showErrorSnackBar = s.shouldShowErrorSnackbar()
+
+                    ReposListUi(repos = s.items, loadMore = showLoadNextPageUi, dispatch = dispatch)
+
+                    val errorMessage = stringResource(R.string.unexpected_error)
+                    if (showErrorSnackBar) {
+                        LaunchedEffect(scaffoldState.snackbarHostState) {
```

Comment:
I'm actually not 100% sure if this works as intended because of the surrounding `if (showErrorSnackBar)` ... It seems to work, but here I don't understand how `LaunchedEffect(key)` learns that the `key = scaffoldState.snackbarHostState` has changed if it surrounded by an `if(showErrorSnackBar)`  ...